packages/client/lib/client.ts

```diff
@@ -54,8 +55,11 @@ export interface EthereumClientOptions {
  */
 export default class EthereumClient {
   public config: Config
+  public chain: Chain
 
-  public services: (FullEthereumService | LightEthereumService)[]
+  public node:
```

Comment:
We've previously used the word "node" before (actually it was what EthereumClient was called before it was renamed, if I remember correctly) so I'd prefer if we can find a more descriptive word here, since "node" is a bit meaningless without more context.

syncmode should already be available in config, so I'm not sure if we need it again. Could you explain your reasoning for combining these properties?  Should we just add "execution" as an optional property on the top level of EthereumClient? Then "services" can also remain as it is.

Reply:
ummm we can use any word other from node as well thats not an issue, basically idea was to use typescript to couple execution with FullEthereumService, syncmode can be dropped. Also a Q, do we need services to be an array?

Reply:
what if we moved execution to FullEthereumService? (i think i remember you suggesting that at first?) because there would never be Light+Execution, so that only leaves Full+Execution

services should be an array to support a full client syncing light peers, unless light sync doesn't go through there, then we could think about it.

Reply:
I think in that case, we can just move execution and services out, as services could be a mix (of light or full) as per your original design , I still would like to keep execution here in client because extracting execution out of services (for e.g. in rpc module) feels a bit weird.(and FullEthereumService still enforces execution in its constructor)

Reply:
nice, cool, just took a look - looks clean!